### PR TITLE
Checking for temp bans

### LINKED ISSUE #2898: Can't make tempbans permanent
# Command bugs

#### Command name

tempban

#### What cog is this command from?

Mod

#### What were you expecting to happen?

To be able to make the tempban permanent later on

#### What actually happened?

Unbanning, hackbanning, or any combination did not remove the user from the tempban list

#### How can we reproduce this issue?

Tempban a user
Unban them
Hackban them
After tempban expires, the user is unbanned despite being unbanned and hackbanned.

### PR DESCRIPTION
**Pull request by rohank98**
### Type

- [x] Bugfix
- [ ] Enhancement
- [ ] New feature

### Description of the changes
Should fix #2898 - checks if user is tempbanned before hackban

### CODE REVIEW

File: redbot/cogs/mod/kickban.py [hunk 1]
```diff
@@ -334,6 +334,15 @@ def remove_processed(ids):
                         user_id=user_id
                     )
 
+        member = namedtuple("Member", "id guild")
+        async with self.settings.guild(guild).current_tempbans() as guild_tempbans:
```
Comment (Drapersniper) [inline redbot/cogs/mod/kickban.py:338]:
```suggestion
        async with self.config.guild(guild).current_tempbans() as guild_tempbans:
```

Please ensure attributes you are using exist.
`self.settings` has been renamed tgo `self.config` a few releases ago.

File: redbot/cogs/mod/kickban.py [hunk 2]
```diff
@@ -334,6 +334,15 @@ def remove_processed(ids):
                         user_id=user_id
                     )
 
+        member = namedtuple("Member", "id guild")
```
Comment (Drapersniper) [inline redbot/cogs/mod/kickban.py:337]:
```suggestion
```
This is not needed if you use the new config methods.

File: redbot/cogs/mod/kickban.py [hunk 3]
```diff
@@ -334,6 +334,15 @@ def remove_processed(ids):
...
+        async with self.settings.guild(guild).current_tempbans() as guild_tempbans:
+            for uid in guild_tempbans.copy():
+                for user_id in user_ids.copy:
+                    if not user_id:
+                        continue
+                await self.settings.member(member(uid, guild)).banned_until.clear()
```
Comment (Drapersniper) [inline redbot/cogs/mod/kickban.py:343]:
```suggestion
                await self.config.member_from_ids(uid, guild.id).banned_until.clear()
```

Comment (Drapersniper) [inline redbot/cogs/mod/kickban.py:343]:
```suggestion
                await self.config.member_from_id(uid, guild.id).banned_until.clear()
```

File: redbot/cogs/mod/kickban.py [hunk 4]
```diff
@@ -334,6 +334,15 @@ def remove_processed(ids):
                         user_id=user_id
                     )
 
+        member = namedtuple("Member", "id guild")
+        async with self.settings.guild(guild).current_tempbans() as guild_tempbans:
+            for uid in guild_tempbans.copy():
+                for user_id in user_ids.copy:
```
Comment (Drapersniper) [inline redbot/cogs/mod/kickban.py:340]: Im not entirely certain what you are trying to do here and in line 341 and 342.

also `.copy` isnt a properpy nor attribute of list did you mean `.copy()` ? if so why are you wanting to use a copy here?

**Review by Drapersniper — CHANGES_REQUESTED** [review]
Let me know if you have any questions.

File: redbot/cogs/mod/kickban.py (hunk 3 above)
Comment (Dav-Git) [reply redbot/cogs/mod/kickban.py:343]: You need to de-draperize this and use self.config instead of self.settings here as well.

**Comment by Drapersniper** [comment]
Another thing to mention ...
If you are going to copy #3635 please credit @Stonedestroyer  as the co-author of this PR/commit.

**Comment by Stonedestroyer** [comment]
Things that attributed to me dropping it as it was in it's unfinished state. 

- Log if tempban failed and why 

I am not sure now what the other reasons considering it was some time ago but this was never tested.

**Comment by Drapersniper** [comment]
Due to the following:

1 - No response in 24 hours.
2 - Copied an existing PR without any context
3 - Did not credit original PR author
4 - PR is currently non function.

I will be closing this PR.
If anyone wishes to pick up #2898 and implement a functional PR please do so.

**Pull request closed without merging**